components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -5929,4 +5957,32 @@ private void setOldOTPRegexValue(
         }
         propertiesFromConnectors.get(otpLengthProperty).setValue(Integer.toString(otpLength));
     }
+
+    private void performConfigCorrectionForPasswordRecoveryConfigs(Connection dbConnection, int tenantId,
+                                       int idpId, List<IdentityProviderProperty> idpProperties) throws SQLException {
+
+        // Enable all recovery options when Recovery.Notification.Password.Enable value is set as enabled.
+        // This keeps functionality consistent with previous API versions for migrating customers.
+        idpProperties.stream().filter(idp -> idp.getName().equals(EMAIL_LINK_PASSWORD_RECOVERY_PROPERTY)).findFirst()
```

Comment:
Shall we switch the attributes order in with equals method to prevent possible null pointers

Reply:
Check other places as well

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/5744/commits/f5e241aeecc63692c94001e3ae24b785eb7cc55e